libs/ledger-live-common/src/api/Evm.ts

```diff
@@ -0,0 +1,213 @@
+
+type AsyncApiFunction = (api: ethers.providers.JsonRpcProvider) => Promise<any>;
+
+/**
+ * Connects to RPC Node
+ */
+async function withApi(
+  currency: CryptoCurrency,
+  execute: AsyncApiFunction
+): Promise<any> {
+  if (!currency?.rpc) {
+    throw new Error("Currency doesn't have an RPC node provided");
+  }
+
+  const provider = new ethers.providers.JsonRpcProvider(currency.rpc);
+  return await execute(provider);
+}
+
+type GetAccountFn = (
+  currency: CryptoCurrency,
+  addr: string
+) => Promise<{ blockHeight: number; balance: BigNumber; nonce: number }>;
+
+/**
+ * Get account balances and nonce
+ */
+export const getAccount: GetAccountFn = async (currency, addr) =>
+  withApi(currency, async (api) => {
+    try {
+      const [balance, nonce, blockHeight] = await Promise.all([
+        getBalance(currency, addr),
+        getTransactionCount(currency, addr),
+        api.getBlockNumber(),
+      ]);
+
+      return {
+        blockHeight,
+        balance: new BigNumber(balance.toString()),
+        nonce,
+      };
+    } catch (e) {
+      console.error(e);
+    }
+  });
+
+/**
+ * Get a transaction by hash
+ */
+export const getTransaction = (
+  currency: CryptoCurrency,
+  hash: string
+): Promise<ethers.providers.TransactionResponse> =>
+  withApi(currency, (api) => {
+    return api.getTransaction(hash);
+  });
+
+/**
+ * Get the balance of an address
+ */
+export const getBalance = (
+  currency: CryptoCurrency,
+  address: string
+): Promise<BigNumber> =>
+  withApi(currency, async (api) => {
+    try {
+      const balance = await api.getBalance(address);
+      return new BigNumber(balance.toString());
+    } catch (e) {
+      return new BigNumber(0);
+    }
+  });
+
+/**
+ * Get account nonce
+ */
+export const getTransactionCount = (
+  currency: CryptoCurrency,
+  addr: string
+): Promise<number> =>
+  withApi(currency, async (api) => {
+    return await api.getTransactionCount(addr);
+  });
+
+/**
+ * Get an estimated gas limit for a transaction
+ */
+export const getGasEstimation = (
+  currency: CryptoCurrency,
+  transaction: EvmTransaction
+): Promise<BigNumber> =>
+  withApi(currency, async (api) => {
+    try {
+      const ethersTransaction = transactionToEthersTransaction(
+        transaction
+      ) as ethers.providers.TransactionRequest;
+      const gasEtimation = await api.estimateGas(ethersTransaction);
+
+      return new BigNumber(gasEtimation.toString());
+    } catch (e) {
+      return new BigNumber(21000);
+    }
+  });
+
+/**
+ * Get an estimation of fees on the network
+ */
+export const getFeesEstimation = (
+  currency: CryptoCurrency
+): Promise<{
+  maxFeePerGas: null | BigNumber;
+  maxPriorityFeePerGas: null | BigNumber;
+  gasPrice: null | BigNumber;
+}> =>
+  withApi(currency, async (api) => {
+    try {
+      const feeData = await api.getFeeData();
+      return {
+        maxFeePerGas: feeData.maxFeePerGas
+          ? new BigNumber(feeData.maxFeePerGas?.toString())
```

Comment:
I think we should test if feeData.maxFeePerGas is null so we can then remove the ? before the toString method

Reply:
Leftover typo, will remove it as we're already testing the existence of `feeData.maxFeePerGas` just a line above. 👌